### PR TITLE
Perf for switching to inbox screen

### PR DESCRIPTION
**Pull request by chrisnojima**
- [x] PERF flag for native so we can focus on components
- [x] rename Inbox cause its using the old name
- [x] defer all outoging rpcs w/ setImmediate, its doing work during mount!
- [x] un-deopt log forwarding

### CODE REVIEW

File: shared/local-debug.native.js [hunk 1]
```diff
@@ -52,6 +54,34 @@ if (__DEV__ && true) {
   config.showAllTrackers = false
 }
 
+if (PERF) {
```
Comment (chrisnojima) [inline shared/local-debug.native.js]: useful for doing component based perf stuff

File: shared/util/forward-logs.js.flow [hunk 2]
```diff
@@ -2,6 +2,39 @@
 export const setupSource = () => {}
 export const setupTarget = () => {}
 
-export const localLog = (...args: any) => {}
-export const localWarn = (...args: any) => {}
-export const localError = (...args: any) => {}
+export const localLog = (
```
Comment (chrisnojima) [inline shared/util/forward-logs.js.flow]: using ...args causes a deopt and is kinda slow

**Comment by chrisnojima** [comment]
there's a big diff on the protocol stuff cause it was never prettified, sorry!

**Comment by chrisnojima** [comment]
@keybase/react-hackers this is ready

File: protocol/package-lock.json [hunk 3]
```diff
@@ -0,0 +1,180 @@
+{
```
Comment (chromakode) [inline protocol/package-lock.json]: Is this from npmv5? Intended to check it in instead of a yarn lock?

File: shared/local-debug.native.js [hunk 4]
```diff
@@ -8,6 +8,8 @@ import {updateDebugConfig} from './actions/dev'
 
 const nativeBridge = NativeModules.KeybaseEngine
 
+const PERF = false
```
Comment (chromakode) [inline shared/local-debug.native.js]: Could you please add a comment to the flag explaining when to use it / what it is appropriate and not appropriate for?

File: shared/util/forward-logs.js.flow (hunk 2 above)
Comment (chromakode) [inline shared/util/forward-logs.js.flow]: This feels a bit gross to me, and I don't think it's necessary if we merge https://github.com/keybase/client/pull/7593

File: shared/util/forward-logs.native.js [hunk 5]
```diff
@@ -18,19 +18,19 @@ function setupSource() {
...
-    logger.info(...args)
+  window.console.log = (a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) => {
+    localLog(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10)
+    logger.info(a1, a2, a3, a4, a5, a6, a7, a8, a9, a10)
   }
 
-  window.console.warn = (...args) => {
-    localWarn(...args)
-    logger.warn(...args)
+  window.console.warn = (a1, a2, a3, a4, a5, a6, a7, a8, a9, a10) => {
```
Comment (chromakode) [inline shared/util/forward-logs.native.js]: I think that warn and error are called infrequently enough that it's not necessary to put in this hack.

File: shared/local-debug.native.js (hunk 4 above)
Comment (chromakode) [inline shared/local-debug.native.js]: Will you always want `__DEV__ === false` in addition to this when using it to get some react-native optimizations?

**Comment by chrisnojima** [comment]
No cause you'll want to perf with dev on and off

**Comment by chrisnojima** [comment]
I don't think this is a hack really. It'll make it faster for dev builds
also

File: shared/util/forward-logs.native.js (hunk 5 above)
Comment (chrisnojima) [inline shared/util/forward-logs.native.js]: this code in general is a hack (overriding console.log) and i think is something we want to move away from and more towards a real logger. but short term this fixes a real perf problem and is totally isolated to the implementation of this passthrough. nothing needs to know about this and it affects nothing else

File: protocol/package-lock.json (hunk 3 above)
Comment (chrisnojima) [inline protocol/package-lock.json]: this is a side effect of the makefile doing an `npm i`. i changed it to use yarn and deleted this file

File: shared/util/forward-logs.native.js (hunk 5 above)
Comment (chrisnojima) [inline shared/util/forward-logs.native.js]: i remove this stuff from the flow files

**Pull request merged**